config/schemas/config.yaml

```diff
@@ -14,6 +14,220 @@ $defs:
     thing:
       $ref: "#/$defs/thing"
     ```
+  affinity:
+    $comment: Real k8s resources might have existing schemas that could be reused as-is
+    additionalProperties:
+      type: object
+    default:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+                - key: node-role.kubernetes.io/control-plane
+                  operator: Exists
+    title: Affinity
+    type: object
+  commonResource:
+    additionalProperties: false
+    description: This is meant to describe the base class if you will, for ck8s resources.
+    properties:
+      affinity:
+        $ref: '#/$defs/affinity'
+      enabled:
+        type: boolean
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      resources:
+        $ref: '#/$defs/resources'
+      tolerations:
+        $ref: '#/$defs/tolerations'
+      topologySpreadConstraints:
+        $ref: '#/$defs/topologySpreadConstraints'
+    required: [] # Shouldn't at least 'enabled' be required?
+    title: Common Resource
+    type: object
+  cpumem:
+    additionalProperties: false
+    properties:
+      cpu:
+        default: 100m
+        pattern: ^[1-9][0-9]*m?$
+        type:
+          - string
+          - integer
+      memory:
+        default: 128Mi
+        pattern: ^[0-9]+(\.[0-9]+)?([KMG]i)?$
+        type:
+          - string
+          - integer
+    title: CPU and Memory specifications
+    type: object
```

Comment:
Can we make the definition for Kubernetes resources follow how they set it up?

- All resources for cpu/memory/storage are defined using the type `Quantity`. (So they have the same validation for cpu and mem.)

- `resources` itself is a `ResourceRequirements` which contains `requests` and `limits`
- `requests` and `limits` are map of string to `Quantity`

Though is this possible to express in the schema, specifically the `map[string]Quantity` type? :sweat_smile:

Reply:
How about now?

```yaml
kubernetesQuantity:
  description: |-
    Used for CPU shares, memory and storage size etc.

    See <https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go>
  regex: ^[-+]?[0-9]+([KMGTPE]i|[eE][0-9]+|[mkMGTPE])?$
  title: Quantity
  type:
    - string
    - number
kubernetesQuantityMap:
  additionalProperties:
    $ref: '#/$defs/kubernetesQuantity'
  type: object
resources:
  additionalProperties: false
  properties:
    limits:
      $ref: '#/$defs/kubernetesQuantityMap'
    requests:
      $ref: '#/$defs/kubernetesQuantityMap'
  title: ResourceRequirements
  type: object
```

Reply:
Looks good :+1:, though I think we should rename ResourceRequeirements to KubernetesResources to keep similar namespacing even though it is called ResourceRequeirements upstream.

Reply:
It's called `kubernetesResourceRequirements` now.